File: Shortest Path/Floyd-Warshall/1956번 - 운동/1956.py
import sys

input = sys.stdin.readline

INF = 987654321

v, e = map(int, input().split())
dist = list([INF] * (v + 1) for _ in range(v+1))
for _ in range(e):
    a, b, c = map(int, input().split())
    dist[a][b] = c;

def solve():
    for k in range(1, v + 1):
        for i in range(1, v + 1):
            for j in range(1, v + 1):
                if dist[i][k] > 0 and dist[k][j] > 0:
                    dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])
                    
    city = list(dist[i][i] for i in range(1, v + 1))
    ans = min(city)
    print(ans if ans != INF else -1)

if __name__ == '__main__':
    solve()

File: Shortest Path/Floyd-Warshall/1956번 - 운동/1956.cpp
#include <iostream>

using namespace std;

#define MAX 401
#define INF 1e9

int v, e;
int dist[MAX][MAX] = { 0, };

void input() {
    cin >> v >> e;
    
    fill_n(&dist[0][0], MAX * MAX, INF);
    for(int i = 0, a, b, c; i < e; i++) {
        cin >> a >> b >> c;
        dist[a][b] = c;
    }
}

void solve() {
    for(int k = 1; k <= v; k++) {
        for(int i = 1; i <= v; i++) {
            for(int j = 1; j <= v; j++) {
                if(dist[i][k] && dist[k][j]) dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j]);
            }
        }
    }
    
    int ans = INF;
    for(int i = 1; i <= v; i++) {
        ans = min(ans, dist[i][i]);
    }
    
    cout << (ans != INF ? ans : -1) << '\n';
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}